traces/metrics.go

```diff
@@ -108,14 +110,14 @@ func Metrics(csvFile string, filenameColumn string, sink MetricsSink) error {
 			}
 
 			return nil
-		})
-		if err != nil {
-			return err
 		}
 
-		// emit metrics
-		err = readLargeCsvFile(csvFile, func(row map[string]string) error {
+		if err := readLargeCsvFile(csvFile,
+			tolerateFaults(csvFile+"#precompute", precomputeMetricsFromRow)); err != nil {
+			return err
+		}
```

Comment:
Ah, my wimpy brain failed to trace this correctly the first two times I read it. I'd recommend using a variable here to reduce the expression complexity.

```suggestion
                var tolerantReader = tolerateFaults(csvFile+"#precompute", precomputeMetricsFromRow) 
		if err := readLargeCsvFile(csvFile, tolerantReader); err != nil {
			return err
		}
```